Modules/_struct.c

```diff
@@ -589,9 +589,10 @@ np_short(_structmodulestate *state, char *p, PyObject *v, const formatdef *f)
     if (get_long(state, v, &x) < 0)
         return -1;
     if (x < SHRT_MIN || x > SHRT_MAX) {
-        PyErr_SetString(state->StructError,
-                        "short format requires " Py_STRINGIFY(SHRT_MIN)
-                        " <= number <= " Py_STRINGIFY(SHRT_MAX));
+        PyErr_Format(state->StructError,
+                     "short format requires %d <= number <= %d",
+                     SHRT_MIN,
+                     SHRT_MAX);
```

Comment:
Convert it to int explicitly:
```suggestion
                     (int)SHRT_MIN, (int)SHRT_MAX);
```

Reply:
@serhiy-storchaka There's no real need, for two reasons: first, `SHRT_MIN` and `SHRT_MAX` already have type `int` (see C99 §5.2.4.2.1p1: "an object of the corresponding type converted according to the integer promotions"); second, even without that, the integer promotions would be applied as normal.

Reply:
The contrast with the unsigned case is that it's always going to be true that C's integer promotions promote `short` to `int`, while `unsigned short` could be promoted to either `int` or `unsigned int` depending on details of the implementation.

Reply:
I just prefer to always explicitly convert arguments of var-arg functions if there are any doubts, because the error can pass silently on some platforms and be catastrophic on others.

Reply:
Sure; I don't feel strongly either way.